src/rgw/rgw_auth_s3.cc

```diff
@@ -919,25 +908,13 @@ AWSv4ComplMulti::ChunkMeta::create_next(CephContext* const cct,
 std::string
 AWSv4ComplMulti::calc_chunk_signature(const std::string& payload_hash) const
 {
-  constexpr size_t algorithm_len = std::strlen(AWS4_HMAC_SHA256_STR);
-  constexpr size_t empty_hash_len = std::strlen(AWS4_EMPTY_PAYLOAD_HASH);
-
-  /* We want to avoid reallocations when concatenating the string_to_sign. */
-  const size_t total_len = algorithm_len + date.length() + \
-    credential_scope.length() + prev_chunk_signature.length() + \
-    empty_hash_len + payload_hash.length() + std::strlen("\n") * 5;
-  const auto string_to_sign = create_n_reserve<std::string>(total_len)
```

Comment:
I think we can safely drop `create_n_reserve` from `rgw_string`. It doesn't seem to be used anywhere else.